tests/test_lab_evaluate.py

```diff
@@ -0,0 +1,329 @@
+
+
+def gen_qa_pairs(odd):
+    i = 1
+    qa_pairs = []
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        qa_pairs.append(
+            {
+                "question_id": i,
+                "score": score,
+                "qna_file": f"category{i}/qna.yaml",
+            }
+        )
+        i = i + 1
+    qa_pairs.append(
+        {
+            "question_id": i,
+            "score": 0.5,
+            "qna_file": f"category{i}/qna.yaml",
+        }
+    )
+    if odd:
+        qa_pairs.append(
+            {
+                "question_id": i + 1,
+                "score": 0.6,
+                "qna_file": f"category{i+1}/qna.yaml",
+            }
+        )
+    return qa_pairs
+
+
+def gen_individual_scores(odd):
+    i = 1
+    individual_scores = {}
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        individual_scores[f"task{i}"] = {"score": score}
+        i = i + 1
+    individual_scores[f"task{i}"] = {"score": 0.5}
+    return individual_scores
+
+
+class TestEvaluate:
+    @patch(
+        "instructlab.model.evaluate.launch_server",
+        return_value=(mock.MagicMock(), "http://127.0.0.1:8000/v1"),
+    )
+    @patch("instructlab.eval.mt_bench.MTBenchEvaluator.gen_answers")
+    @patch(
+        "instructlab.eval.mt_bench.MTBenchEvaluator.judge_answers",
+        return_value=(1.5001, [{}, {}], [1.002, 2]),
+    )
+    def test_evaluate_mt_bench(
+        self,
+        judge_answers_mock,
+        gen_answers_mock,
+        launch_server_mock,
+    ):
+        runner = CliRunner()
+        with runner.isolated_filesystem():
+            result = runner.invoke(
+                lab.ilab,
+                [
+                    "--config=DEFAULT",
+                    "model",
+                    "evaluate",
+                    "--benchmark",
+                    "mt_bench",
+                    "--model",
+                    "models/instructlab/granite-7b-lab",
+                    "--judge-model",
+                    "models/instructlab/merlinite-7b-lab",
+                ],
+            )
+            judge_answers_mock.assert_called_once()
+            gen_answers_mock.assert_called_once()
+            assert launch_server_mock.call_count == 2
+            assert result.exit_code == 0
+            assert (
+                result.output
+                == """Generating answers...
+Evaluating answers...
+# SKILL EVALUATION REPORT
+
+## MODEL
+models/instructlab/granite-7b-lab
+
+### AVERAGE:
+1.5 (across 2)
+
+### TURN ONE:
+1.0
+
+### TURN TWO:
+2
+"""
```

Comment:
The inline text without same indention makes the code harder to follow. I like to use [textwrap.dedent](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) to have nicer indentions

```python
def test():
    expected = textwrap.dedent(
        """\
        Generating answers...
        Evaluating answers...
        ...
        """
    )
    assert result == expected
```

Reply:
Thanks for the suggestion!